presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -0,0 +1,481 @@
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.metastore.MetaStoreUtils;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.ql.io.SymlinkTextInputFormat;
+import org.apache.hadoop.mapred.FileInputFormat;
+import org.apache.hadoop.mapred.FileSplit;
+import org.apache.hadoop.mapred.InputFormat;
+import org.apache.hadoop.mapred.InputSplit;
+import org.apache.hadoop.mapred.JobConf;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.facebook.presto.hadoop.HadoopFileStatus.isDirectory;
+import static com.facebook.presto.hadoop.HadoopFileStatus.isFile;
+import static com.facebook.presto.hive.HiveBucketing.HiveBucket;
+import static com.facebook.presto.hive.HiveType.getSupportedHiveType;
+import static com.facebook.presto.hive.HiveUtil.getInputFormat;
+import static com.facebook.presto.hive.HiveUtil.isSplittable;
+import static com.facebook.presto.hive.UnpartitionedPartition.isUnpartitioned;
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+public class BackgroundHiveSplitLoader
+        implements HiveSplitLoader
+{
+    public static final String FORCE_LOCAL_SCHEDULING = "force_local_scheduling";
+
+    private final String connectorId;
+    private final Table table;
+    private final Optional<HiveBucket> bucket;
+    private final HdfsEnvironment hdfsEnvironment;
+    private final NamenodeStats namenodeStats;
+    private final DirectoryLister directoryLister;
+    private final DataSize maxSplitSize;
+    private final int maxPartitionBatchSize;
+    private final DataSize maxInitialSplitSize;
+    private final boolean recursiveDirWalkerEnabled;
+    private final boolean forceLocalScheduling;
+    private final Executor executor;
+    private final ConnectorSession session;
+    private final AtomicInteger outstandingTasks = new AtomicInteger();
+    private final Queue<HivePartitionMetadata> partitions = new ConcurrentLinkedQueue<>();
+    private final Deque<HiveFileIterator> fileIterators = new ConcurrentLinkedDeque<>();
+    private final AtomicInteger remainingInitialSplits;
+
+    private HiveSplitSource hiveSplitSource;
+    private volatile boolean stopped;
+
+    public BackgroundHiveSplitLoader(String connectorId,
```

Comment:
Is there a way to add that to the style checker? I never remember where to put the first arg